bson/json_util.py

```diff
@@ -448,6 +448,20 @@ def with_options(self, **kwargs: Any) -> JSONOptions:
 .. versionadded:: 3.4
 """
 
+# Encoders for BSON types
+_encoders = {
+    5: lambda obj, json_options: _encode_binary(obj, obj.subtype, json_options),  # Binary
+    7: lambda obj, json_options: {"$oid": str(obj)},  # noqa: ARG005 ObjectId
+    9: lambda obj, json_options: _encode_datetimems(obj, json_options),  # DatetimeMS
```

Comment:
A number of these can be simplified to remove a redundant intermediate call:
```
 9: _encode_datetimems,
 13: _encode_code,
 18: _encode_int64,
```

Reply:
`_encode_code` and `_encode_int64` are fine to inline, `_encode_datetimems` would be significantly less readable with its three distinct return paths.

Reply:
Uh? I'm saying the code can literally be:
```python
 9: _encode_datetimems,
 13: _encode_code,
 18: _encode_int64,
```

Reply:
Oh sorry, I thought you meant inline those functions into the lambda itself.

Reply:
That works, just have to move the definition of `_encoders` down to after all of the encoding methods are defined.

Reply:
Oh I see now thanks. I also agree that embedding complex logic into a lambda would not be ideal.